scripts/modules/MsIndustryLinks/templates/ingest/IngestionWorkflow.psm1

```diff
@@ -78,31 +80,234 @@ function New-LogicAppIngestionWorkflow {
         [object] $DataSinkConfig
     )
 
-    $definitionFileSuffix = $(if ($DataSinkConfig.upsert) { "upsert" } else { "insert" })
+    $dataSinkType = $DataSinkConfig.type.ToLower()
 
+    $definitionFileSuffix = $(if ($dataSinkType -eq "dataverse") { if ($DataSinkConfig.upsert) { "_upsert" } else { "_insert" } } else { "" })
+    $definition = Get-Content "$PSScriptRoot/templates/ingest/$dataSinkType/logicapp_$($dataSinkType)$($definitionFileSuffix).json" | ConvertFrom-Json
     $baseTemplate = Get-Content $PSScriptRoot/templates/logicapp_base.json | ConvertFrom-Json
-    $definition = Get-Content $PSScriptRoot/templates/ingest/logicapp_dataverse_${definitionFileSuffix}.json | ConvertFrom-Json
 
-    $definition.actions.For_each_item.actions.Ingest_record.inputs.body = $DataSinkConfig.mapping
+    switch ($dataSinkType) {
+        "customconnector" {
+            $apiName = $DataSinkConfig.properties.name
+            $apiId = Get-LogicAppApiId -DataSourceType $dataSinkType -ApiName $apiName -IsCustomConnectorCertified $DataSinkConfig.isCertified
+
+            Set-LogicAppCustomConnectorDataSinkConfiguration -DataSinkConfig $DataSinkConfig -Definition $definition | Out-Null
+        }
+        "dataverse" {
+            $apiName = Get-ApiName -DataSourceType $dataSinkType
+            $apiId = Get-LogicAppApiId -DataSourceType $dataSinkType -ApiName $apiName
```

Comment:
Where is this new `Get-LogicAppApiId` function? I don't see it in the file.

Reply:
Its a shared function from the DataSourceWorkflow module. It will be one that we split out into a module for shared functions and logic that we discussed earlier, along with Get-ApiName.